app/helpers/orchid/date_helper.rb

```diff
@@ -39,13 +35,34 @@ def date_filter options
     end
     options.delete("date_from")
     options.delete("date_to")
-    return [options, from, to]
+    [options, from, to]
+  end
+
+  def date_format(date)
+    y, m, d = date.split("-")
+
+    # Fix numbers out of range
+    y = 1 if y.to_i < 1
```

Comment:
should this use the year start and year end range that orchid apps specify in the config?  that way if you filled in a bad lewis and clark date, for example, it would start in 1803?

Reply:
That was my first inclination, but I hesitated because we won't know if the date being formatted is the first or second date. I wondered if the second date might somehow end up getting set earlier than the first and break things. But I suppose we could add another parameter to the method to indicate that. Sound like a good approach?

Reply:
Ah good point. I think that another parameter makes sense!

Reply:
Functionality added. Take a look :)

Reply:
Code looks a lot tighter!  I like it